File: Assets/Convergence Assets/Scripts/Const.js
static var greenButton : GUIStyle;
static var redButton : GUIStyle;
static var greenLabel : GUIStyle;
static var redLabel : GUIStyle;
private var done : boolean = false;
static var noiseMagnitude : float = 0.2;
static var dampFactor : float = 0.95;
var useNoise : boolean = false;
var enableProfiling : boolean = false;
static var controlBusy : boolean = false;
static var usesNoise : boolean;
static var menuScrolling : boolean = false;
var dataBaseName : String = "///unitygames/testdata.sqlite";
static var dataBase : String;
static var bigLabel : GUIStyle;
static var fixLabel : GUIStyle;
static var centerLabel : GUIStyle;
static var littleCenterLabel : GUIStyle;
static var littleBlackLabel : GUIStyle;
static var bigRedLabel : GUIStyle;
static var greenCenterLabel : GUIStyle;
static var rightLabel : GUIStyle;
static var bigToggle : GUIStyle;
static var buttonToggle : GUIStyle;
static var realToggle : GUIStyle;
static var bigInput : GUIStyle;
static var grayStyle : GUIStyle;
static var grayButton : GUIStyle;
static var errorStyle : GUIStyle;
static var solidWindowStyle : GUIStyle;
var bigFont : Font;
var inputTexture : Texture;
var grayTexture : Texture;
var errorTexture : Texture;

function Awake()
{
	Debug.Log("Awaking in const");
	Const.dataBase = dataBaseName;
	greenButton = new GUIStyle();
	redButton = new GUIStyle();
	greenLabel = new GUIStyle();
	redLabel = new GUIStyle();
	usesNoise = useNoise;
	if(enableProfiling)
	{
		UnityEngine.Profiling.Profiler.logFile = "mylog.log";
		UnityEngine.Profiling.Profiler.enabled = true;
	}
}

private var gcount : int = 0;

function OnGUI()
{
	if(++gcount < 3)
	{
		//Debug.Log("in const OnGUI " + gcount);
	}
	if(!done)
	{
		greenButton = new GUIStyle(GUI.skin.button);
		redButton = new GUIStyle(GUI.skin.button);
		redButton.normal.textColor = new Color(1.0, 0.0, 0.0, 1.0);
		greenButton.normal.textColor = new Color(0.0, 1.0, 0.0, 1.0);
		greenLabel = new GUIStyle(GUI.skin.label);
		redLabel = new GUIStyle(GUI.skin.label);
		redLabel.normal.textColor = new Color(1.0, 0.0, 0.0, 1.0);
		greenLabel.normal.textColor = new Color(0.0, 1.0, 0.0, 1.0);
		bigLabel = new GUIStyle(GUI.skin.label);
		bigLabel.font = bigFont;
		bigLabel.alignment = TextAnchor.UpperLeft;
		fixLabel = new GUIStyle(GUI.skin.label);
		fixLabel.font = bigFont;
		fixLabel.alignment = TextAnchor.UpperLeft;
		fixLabel.normal.textColor = new Color(0.0, 0.0, 0.0, 1.0);
		
		centerLabel = new GUIStyle(GUI.skin.label);
		centerLabel.font = bigFont;
		centerLabel.alignment = TextAnchor.UpperCenter;
		centerLabel.normal.textColor = new Color(0.0, 0.0, 0.0, 1.0);
		
		littleCenterLabel = new GUIStyle(GUI.skin.label);
		littleCenterLabel.alignment = TextAnchor.UpperCenter;
		
		littleBlackLabel = new GUIStyle(GUI.skin.label);
		littleBlackLabel.normal.textColor = new Color(0.0, 0.0, 0.0, 1.0);
		
		greenCenterLabel = new GUIStyle(GUI.skin.label);
		greenCenterLabel.font = bigFont;
		greenCenterLabel.alignment = TextAnchor.UpperCenter;
		greenCenterLabel.normal.textColor = new Color(0.0, 0.8, 0.0, 1.0);
		
		bigRedLabel = new GUIStyle(GUI.skin.label);
		bigRedLabel.font = bigFont;
		bigRedLabel.alignment = TextAnchor.UpperLeft;
		bigRedLabel.normal.textColor = new Color(1.0, 0.2, 0.2, 1.0);
		
		rightLabel = new GUIStyle(GUI.skin.label);
		rightLabel.font = bigFont;
		rightLabel.alignment = TextAnchor.UpperRight;
		bigToggle = new GUIStyle(GUI.skin.button);
		bigToggle.font = bigFont;
		bigToggle.stretchWidth = true;
		bigToggle.stretchHeight = true;
		bigToggle.alignment = TextAnchor.UpperCenter;
		
		buttonToggle = new GUIStyle(GUI.skin.button);
		buttonToggle.font = bigFont;
		buttonToggle.alignment = TextAnchor.UpperCenter;
		
		realToggle = new GUIStyle(GUI.skin.toggle);
		realToggle.font = bigFont;
		realToggle.alignment = TextAnchor.UpperLeft;
		bigInput = new GUIStyle(GUI.skin.label);
		bigInput.font = bigFont;
		bigInput.alignment = TextAnchor.UpperLeft;
		bigInput.normal.background = inputTexture;
		
		grayStyle = new GUIStyle(GUI.skin.label);
		grayStyle.font = bigFont;
		grayStyle.alignment = TextAnchor.UpperLeft;
		grayStyle.normal.background = grayTexture;

		grayButton = new GUIStyle(GUI.skin.button);
		grayButton.font = bigFont;
		grayButton.alignment = TextAnchor.UpperCenter;
		grayButton.normal.background = grayTexture;

		errorStyle = new GUIStyle(GUI.skin.label);
		errorStyle.font = bigFont;
		errorStyle.alignment = TextAnchor.UpperLeft;
		errorStyle.normal.background = errorTexture;
		
		solidWindowStyle = new GUIStyle(GUI.skin.label);
		solidWindowStyle.normal.background = inputTexture;
	
		done = true;
	}
}

private var upcount : int = 0;
function Update ()
{
	if(++upcount < 3)
	{
		//Debug.Log("in const update");
	}
	if (Input.GetKey ("escape"))
	{
		Debug.Log("heard escape key in Const");
		Application.Quit();
		//Debug.Log("if we quit, why are we still here?");
	}
}

File: Assets/Convergence Assets/Scripts/Const.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Const : MonoBehaviour
{
    public static GUIStyle greenButton;
    public static GUIStyle redButton;
    public static GUIStyle greenLabel;
    public static GUIStyle redLabel;
    private bool done;
    public static float noiseMagnitude;
    public static float dampFactor;
    public bool useNoise;
    public bool enableProfiling;
    public static bool controlBusy;
    public static bool usesNoise;
    public static bool menuScrolling;
    public string dataBaseName;
    public static string dataBase;
    public static GUIStyle bigLabel;
    public static GUIStyle fixLabel;
    public static GUIStyle centerLabel;
    public static GUIStyle littleCenterLabel;
    public static GUIStyle littleBlackLabel;
    public static GUIStyle bigRedLabel;
    public static GUIStyle greenCenterLabel;
    public static GUIStyle rightLabel;
    public static GUIStyle bigToggle;
    public static GUIStyle buttonToggle;
    public static GUIStyle realToggle;
    public static GUIStyle bigInput;
    public static GUIStyle grayStyle;
    public static GUIStyle grayButton;
    public static GUIStyle errorStyle;
    public static GUIStyle solidWindowStyle;
    public Font bigFont;
    public Texture2D inputTexture; //Stauffer - change these to Texture2D
    public Texture2D grayTexture;
    public Texture2D errorTexture;
    public virtual void Awake()
    {
        Debug.Log("Awaking in const");
        Const.dataBase = this.dataBaseName;
        Const.greenButton = new GUIStyle();
        Const.redButton = new GUIStyle();
        Const.greenLabel = new GUIStyle();
        Const.redLabel = new GUIStyle();
        Const.usesNoise = this.useNoise;
        if (this.enableProfiling)
        {
            UnityEngine.Profiling.Profiler.logFile = "mylog.log";
            UnityEngine.Profiling.Profiler.enabled = true;
        }
    }

    private int gcount;
    public virtual void OnGUI()
    {
        if (++this.gcount < 3)
        {
        }
         //Debug.Log("in const OnGUI " + gcount);
        if (!this.done)
        {
            Const.greenButton = new GUIStyle(GUI.skin.button);
            Const.redButton = new GUIStyle(GUI.skin.button);
            Const.redButton.normal.textColor = new Color(1f, 0f, 0f, 1f);
            Const.greenButton.normal.textColor = new Color(0f, 1f, 0f, 1f);
            Const.greenLabel = new GUIStyle(GUI.skin.label);
            Const.redLabel = new GUIStyle(GUI.skin.label);
            Const.redLabel.normal.textColor = new Color(1f, 0f, 0f, 1f);
            Const.greenLabel.normal.textColor = new Color(0f, 1f, 0f, 1f);
            Const.bigLabel = new GUIStyle(GUI.skin.label);
            Const.bigLabel.font = this.bigFont;
            Const.bigLabel.alignment = TextAnchor.UpperLeft;
            Const.fixLabel = new GUIStyle(GUI.skin.label);
            Const.fixLabel.font = this.bigFont;
            Const.fixLabel.alignment = TextAnchor.UpperLeft;
            Const.fixLabel.normal.textColor = new Color(0f, 0f, 0f, 1f);
            Const.centerLabel = new GUIStyle(GUI.skin.label);
            Const.centerLabel.font = this.bigFont;
            Const.centerLabel.alignment = TextAnchor.UpperCenter;
            Const.centerLabel.normal.textColor = new Color(0f, 0f, 0f, 1f);
            Const.littleCenterLabel = new GUIStyle(GUI.skin.label);
            Const.littleCenterLabel.alignment = TextAnchor.UpperCenter;
            Const.littleBlackLabel = new GUIStyle(GUI.skin.label);
            Const.littleBlackLabel.normal.textColor = new Color(0f, 0f, 0f, 1f);
            Const.greenCenterLabel = new GUIStyle(GUI.skin.label);
            Const.greenCenterLabel.font = this.bigFont;
            Const.greenCenterLabel.alignment = TextAnchor.UpperCenter;
            Const.greenCenterLabel.normal.textColor = new Color(0f, 0.8f, 0f, 1f);
            Const.bigRedLabel = new GUIStyle(GUI.skin.label);
            Const.bigRedLabel.font = this.bigFont;
            Const.bigRedLabel.alignment = TextAnchor.UpperLeft;
            Const.bigRedLabel.normal.textColor = new Color(1f, 0.2f, 0.2f, 1f);
            Const.rightLabel = new GUIStyle(GUI.skin.label);
            Const.rightLabel.font = this.bigFont;
            Const.rightLabel.alignment = TextAnchor.UpperRight;
            Const.bigToggle = new GUIStyle(GUI.skin.button);
            Const.bigToggle.font = this.bigFont;
            Const.bigToggle.stretchWidth = true;
            Const.bigToggle.stretchHeight = true;
            Const.bigToggle.alignment = TextAnchor.UpperCenter;
            Const.buttonToggle = new GUIStyle(GUI.skin.button);
            Const.buttonToggle.font = this.bigFont;
            Const.buttonToggle.alignment = TextAnchor.UpperCenter;
            Const.realToggle = new GUIStyle(GUI.skin.toggle);
            Const.realToggle.font = this.bigFont;
            Const.realToggle.alignment = TextAnchor.UpperLeft;
            Const.bigInput = new GUIStyle(GUI.skin.label);
            Const.bigInput.font = this.bigFont;
            Const.bigInput.alignment = TextAnchor.UpperLeft;
            Const.bigInput.normal.background = this.inputTexture;
            Const.grayStyle = new GUIStyle(GUI.skin.label);
            Const.grayStyle.font = this.bigFont;
            Const.grayStyle.alignment = TextAnchor.UpperLeft;
            Const.grayStyle.normal.background = this.grayTexture;
            Const.grayButton = new GUIStyle(GUI.skin.button);
            Const.grayButton.font = this.bigFont;
            Const.grayButton.alignment = TextAnchor.UpperCenter;
            Const.grayButton.normal.background = this.grayTexture;
            Const.errorStyle = new GUIStyle(GUI.skin.label);
            Const.errorStyle.font = this.bigFont;
            Const.errorStyle.alignment = TextAnchor.UpperLeft;
            Const.errorStyle.normal.background = this.errorTexture;
            Const.solidWindowStyle = new GUIStyle(GUI.skin.label);
            Const.solidWindowStyle.normal.background = this.inputTexture;
            this.done = true;
        }
    }

    private int upcount;
    public virtual void Update()//Debug.Log("if we quit, why are we still here?");
    {
        if (++this.upcount < 3)
        {
        }
         //Debug.Log("in const update");
        if (Input.GetKey("escape"))
        {
            Debug.Log("heard escape key in Const");
            Application.Quit();
        }
    }

    public Const()
    {
        this.dataBaseName = "///unitygames/testdata.sqlite";
    }

    static Const()
    {
        Const.noiseMagnitude = 0.2f;
        Const.dampFactor = 0.95f;
    }

}